### PR TITLE
Upgrade to camel 3.0.0 and quarkus 1.0.0.Final

### LINKED ISSUE #319: Remove MicroProfile Metrics extension workarounds
Once the version of MicroProfile Metrics API is aligned between Quarkus and Camel, we can remove the [workarounds](https://github.com/apache/camel-quarkus/tree/master/extensions/microprofile-metrics/runtime/src/main/java/org/apache/camel/quarkus/component/microprofile/metrics/runtime/patch) that I had to add in order to maintain compatibility between Camel 3.0.0-RC3 & Quarkus 0.26.

### LINKED ISSUE #469: Upgrade to camel  3.0.0
We can use staging repo till the release is out:

    https://repository.apache.org/content/repositories/orgapachecamel-1164/

### LINKED ISSUE #470: Upgrade to quarkus 1.0.0.Final

### PR DESCRIPTION
**Pull request by lburgazzoli**
Fixes #470
Fixes #319
Fixes #469

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@johnpoth looks like we need to add a test for fhir r5
@aldettinger the pdf extensions now fails in native mode, I've disabled tests for the moment, see https://github.com/apache/camel-quarkus/pull/471/files#diff-7cc83a1071a5196fb78d4f99b1bb014cR67-R93

File: extensions/fhir/runtime/pom.xml [hunk 1]
```diff
@@ -74,6 +70,11 @@
             <groupId>org.jboss.logging</groupId>
             <artifactId>commons-logging-jboss-logging</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>com.oracle.substratevm</groupId>
+            <artifactId>svm</artifactId>
```
Comment (ppalaga) [inline extensions/fhir/runtime/pom.xml]: Is this really needed in FHIR? I see no substitutions or anything that would require this dep.

Comment (ppalaga) [reply extensions/fhir/runtime/pom.xml]: > I see no substitutions

I see them now, sorry. Had to hit F5 in Eclipse.

File: integration-tests/pdf/pom.xml [hunk 2]
```diff
@@ -64,10 +64,37 @@
   </build>
   <profiles>
     <profile>
-      <id>native</id>
+      <!--
+      TODO: fix and re-enable native tests
+
```
Comment (ppalaga) [inline integration-tests/pdf/pom.xml]: An issue would be nice so that we do not forget.

Comment (aldettinger) [reply integration-tests/pdf/pom.xml]: Ok, I've created https://github.com/apache/camel-quarkus/issues/472 not to forget.

**Comment by lburgazzoli** [comment]
ok to test

**Comment by lburgazzoli** [comment]
ok to test

**Comment by lburgazzoli** [comment]
ok to test

**Pull request merged**